File: modules/coupon-domain/src/main/java/com/huythanh0x/udemycoupons/dto/IncentivesDTO.java
package com.huythanh0x.udemycoupons.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO for course incentives (certificate, lifetime access, etc.).
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class IncentivesDTO {
    private String videoContentLength; // e.g., "2.5 hours"
    private Integer numArticles;
    private Integer numQuizzes;
    private Integer numPracticeTests;
    private Integer numCodingExercises;
    private Boolean hasLifetimeAccess;
    private String devicesAccess; // e.g., "Access on mobile and TV"
    private Boolean hasAssignments;
    private Boolean hasCertificate;
    private Boolean hasClosedCaptions;
}

File: modules/coupon-domain/src/main/kotlin/com/thanh0x/coursedeal/dto/IncentivesDTO.kt
package com.thanh0x.coursedeal.dto

/**
 * DTO for course incentives (certificate, lifetime access, etc.).
 */
data class IncentivesDTO(
    // e.g., "2.5 hours"
    val videoContentLength: String? = null,
    val numArticles: Int? = null,
    val numQuizzes: Int? = null,
    val numPracticeTests: Int? = null,
    val numCodingExercises: Int? = null,
    val hasLifetimeAccess: Boolean? = null,
    // e.g., "Access on mobile and TV"
    val devicesAccess: String? = null,
    val hasAssignments: Boolean? = null,
    val hasCertificate: Boolean? = null,
    val hasClosedCaptions: Boolean? = null,
)
